### ISSUE TITLE
Add constraints on voltage and frequency (Trac #159)

### ISSUE THREAD

**Issue by cfobel** [wheeler-microfluidics/microdrop#163]
There is nothing currently stopping users from entering voltages and frequencies that are beyond the capabilities of the system. It would be useful to be able to set limitations on these parameters that would prevent people from accidentally damaging their devices/hardware.

These values could be stored in the config file as app options.

Migrated from http://microfluidics.utoronto.ca/microdrop/ticket/159
```json
{
    "status": "closed", 
    "changetime": "2014-11-06T22:22:09", 
    "description": "There is nothing currently stopping users from entering voltages and frequencies that are beyond the capabilities of the system. It would be useful to be able to set limitations on these parameters that would prevent people from accidentally damaging their devices/hardware.\n\nThese values could be stored in the config file as app options.", 
    "reporter": "ryan", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1415312529456030", 
    "component": "microdrop", 
    "summary": "Add constraints on voltage and frequency", 
    "priority": "minor", 
    "keywords": "", 
    "version": "0.1", 
    "time": "2014-10-28T21:29:40", 
    "milestone": "Control board plugin 1.0", 
    "owner": "", 
    "type": "enhancement"
}
```

**Comment by cfobel** [comment]
Trac update at 20141103T17:33:35: 
* ryan commented:

> Rather than storing these limits as app options, I think it makes more sense to store them on the control board EEPROM since they may vary on a per system basis (depending on the amplifier, etc.). I've created a control board firmware feature request: [http://microfluidics.utoronto.ca/trac/dropbot/ticket/66#ticket #70].
> 
> The plugin can therefore validate voltage and frequency settings against the currently connected system. If no system is present, don't perform any validation beyond checking for values > 0.
> 
> Also changing the milestone to for this ticket to [http://microfluidics.utoronto.ca/microdrop/milestone/Control%20board%20plugin%201.0 Control board plugin 1.0].


* ryan changed milestone from "Microdrop 1.0" to "Control board plugin 1.0"

**Comment by cfobel** [comment]
Trac update at 20141106T22:22:09: 
* ryan commented:

> This is fixed as of commit [changeset:80b1d1f/dmf_control_board_plugin].


* ryan changed resolution from "" to "fixed"
* ryan changed status from "new" to "closed"
